src/builder.rs

```diff
@@ -221,6 +221,7 @@ pub struct Builder {
     recipients: Vec<RecipientInfo>,
     flags: Flags,
     anchor: Anchor,
+    value_balance: i64,
```

Comment:
@nuttycom and I: Why cache this and update it in `add_spend`/`add_recipient`, rather than recomputing it in the `value_balance` method?